{
  "cve_id": "CVE-2021-27213",
  "github_data": {
    "repository": "cvandeplas/pystemon",
    "fix_commit": "dbeb87afefdb63de2f4cff69b6f10c5965d14b54",
    "related_commits": [
      "dbeb87afefdb63de2f4cff69b6f10c5965d14b54",
      "dbeb87afefdb63de2f4cff69b6f10c5965d14b54"
    ],
    "patch_url": "https://github.com/cvandeplas/pystemon/commit/dbeb87afefdb63de2f4cff69b6f10c5965d14b54.patch",
    "fix_commit_details": {
      "sha": "dbeb87afefdb63de2f4cff69b6f10c5965d14b54",
      "commit_date": "2021-02-13T17:36:38Z",
      "author": {
        "login": "Asjidkalam",
        "type": "User",
        "stats": null
      },
      "commit_message": {
        "title": "Fixed code execution bug using SafeLoader()",
        "length": 43,
        "has_description": false,
        "references_issue": true
      },
      "stats": {
        "total": 6,
        "additions": 3,
        "deletions": 3
      },
      "files": [
        {
          "filename": "pystemon/config.py",
          "status": "modified",
          "additions": 3,
          "deletions": 3,
          "patch": "@@ -304,9 +304,9 @@ def _load_yamlconfig(self, configfile):\n             if self._recent_pyyaml():\n                 # https://github.com/yaml/pyyaml/wiki/PyYAML-yaml.load(input)-Deprecation\n                 # only for 5.1+\n-                yamlconfig = yaml.load(open(configfile), Loader=yaml.FullLoader)\n+                yamlconfig = yaml.load(open(configfile), Loader=yaml.SafeLoader)\n             else:\n-                yamlconfig = yaml.load(open(configfile))\n+                yamlconfig = yaml.safe_load(open(configfile))\n         except yaml.YAMLError as exc:\n             logger.error(\"Error in configuration file {0}:\".format(configfile))\n             if hasattr(exc, 'problem_mark'):\n@@ -315,7 +315,7 @@ def _load_yamlconfig(self, configfile):\n         for includes in yamlconfig.get(\"includes\", []):\n             try:\n                 logger.debug(\"loading include '{0}'\".format(includes))\n-                yamlconfig.update(yaml.load(open(includes)))\n+                yamlconfig.update(yaml.safe_load(open(includes)))\n             except Exception as e:\n                 raise PystemonConfigException(\"failed to load '{0}': {1}\".format(includes, e))\n         return yamlconfig"
        }
      ],
      "file_patterns": {
        "security_files": 0,
        "config_files": 1,
        "dependency_files": 0,
        "test_files": 0,
        "unique_directories": 1,
        "max_directory_depth": 1
      },
      "context": {
        "surrounding_commits": [
          {
            "sha": "7e3665674f60250c198b23ad4b87d4393e8e65bb",
            "date": "2022-05-31T18:57:25Z",
            "author_login": "cvandeplas"
          },
          {
            "sha": "37023e935c88c0efc3b158fe3153e2f7aa233671",
            "date": "2022-05-31T17:54:50Z",
            "author_login": "Gurulhu"
          },
          {
            "sha": "2743b7a22120319be0e9b222813e3bc2c5024802",
            "date": "2021-11-23T20:13:12Z",
            "author_login": "cvandeplas"
          },
          {
            "sha": "795903d800825c5b7404d055aa49bb14e1473f2a",
            "date": "2021-11-23T19:11:52Z",
            "author_login": "timgates42"
          },
          {
            "sha": "bedb6a08f32da4b93dc55c18bfd46b46afc1a04e",
            "date": "2021-08-25T11:02:49Z",
            "author_login": "cvandeplas"
          }
        ]
      }
    }
  },
  "vulnerability_details": {
    "cvss_score": 9.8,
    "cvss_vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
    "cwe_id": "CWE-502",
    "description": "config.py in pystemon before 2021-02-13 allows code execution via YAML deserialization because SafeLoader and safe_load are not used.",
    "attack_vector": "NETWORK",
    "attack_complexity": "LOW"
  },
  "temporal_data": {
    "published_date": "2021-02-14T04:15:12.837",
    "last_modified": "2024-11-21T05:57:36.437",
    "fix_date": "2021-02-13T17:36:38Z"
  },
  "references": [
    {
      "url": "https://github.com/cvandeplas/pystemon/commit/dbeb87afefdb63de2f4cff69b6f10c5965d14b54",
      "source": "cve@mitre.org",
      "tags": [
        "Patch",
        "Third Party Advisory"
      ]
    },
    {
      "url": "https://www.huntr.dev/bounties/1-other-pystemon/",
      "source": "cve@mitre.org",
      "tags": [
        "Exploit",
        "Third Party Advisory"
      ]
    },
    {
      "url": "https://github.com/cvandeplas/pystemon/commit/dbeb87afefdb63de2f4cff69b6f10c5965d14b54",
      "source": "af854a3a-2127-422b-91ae-364da2661108",
      "tags": [
        "Patch",
        "Third Party Advisory"
      ]
    },
    {
      "url": "https://www.huntr.dev/bounties/1-other-pystemon/",
      "source": "af854a3a-2127-422b-91ae-364da2661108",
      "tags": [
        "Exploit",
        "Third Party Advisory"
      ]
    }
  ],
  "collection_metadata": {
    "collected_at": "2025-01-11T23:01:16.785601",
    "processing_status": "raw"
  },
  "repository_context": {
    "name": "pystemon",
    "owner": "cvandeplas",
    "created_at": "2012-10-05T08:35:52Z",
    "updated_at": "2024-08-12T19:11:13Z",
    "pushed_at": "2022-06-02T13:46:57Z",
    "size": 400,
    "stars": 334,
    "forks": 223,
    "open_issues": 14,
    "watchers": 334,
    "has_security_policy": false,
    "default_branch": "master",
    "protected_branches": [],
    "languages": {
      "Python": 101967,
      "Dockerfile": 1104
    },
    "commit_activity": {
      "total_commits_last_year": 0,
      "avg_commits_per_week": 0,
      "days_active_last_year": 0
    },
    "security_features": {
      "has_security_policy": false,
      "has_protected_branches": false,
      "has_wiki": true,
      "has_issues": true,
      "allow_forking": true,
      "is_template": false,
      "license": "agpl-3.0"
    },
    "collected_at": "2025-01-14T18:10:23.727402"
  }
}